### ISSUE TITLE
How I allow user to register without items in cart?

### ISSUE THREAD

**Issue by vivazzi** [awesto/django-shop#601]
Now I need to allow user to register without items in cart, but if I use AuthFormsView (by default) I see warning: "Unable to proceed to the checkout page without items in the cart."

I try to use custom AuthFormsView like this:
```
class CustomAuthFormsView(GenericAPIView):
    serializer_class = None
    form_class = None

    def post(self, request, *args, **kwargs):
        # if request.customer.is_visitor():
        #     errmsg = _("Unable to proceed to the checkout page without items in the cart.")
        #     errors = {NON_FIELD_ERRORS: errmsg}
        #     return Response(errors, status=status.HTTP_400_BAD_REQUEST)
        form = self.form_class(data=request.data, instance=request.customer)
        if form.is_valid():
            form.save(request=request)
            return Response(form.data, status=status.HTTP_200_OK)
        return Response(dict(form.errors), status=status.HTTP_400_BAD_REQUEST)
```

But I get error:
```
Traceback (most recent call last):
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 42, in inner
    response = get_response(request)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/rest_framework/views.py", line 477, in dispatch
    response = self.handle_exception(exc)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/rest_framework/views.py", line 437, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/rest_framework/views.py", line 474, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/vivazzi/projects/lonx/spec/forms.py", line 21, in post
    form = self.form_class(data=request.data, instance=request.customer)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/shop/forms/auth.py", line 55, in __init__
    super(RegisterUserForm, self).__init__(data=data, instance=instance, *args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/djng/forms/angular_model.py", line 41, in __init__
    super(NgModelFormMixin, self).__init__(data, *args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/djng/forms/angular_validation.py", line 17, in __init__
    super(NgFormValidationMixin, self).__init__(*args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/djng/forms/angular_base.py", line 252, in __init__
    super(NgFormBaseMixin, self).__init__(data=data, *args, **kwargs)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/forms/models.py", line 282, in __init__
    object_data = model_to_dict(instance, opts.fields, opts.exclude)
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/forms/models.py", line 87, in model_to_dict
    opts = instance._meta
  File "/home/vivazzi/projects/lonx/env/local/lib/python2.7/site-packages/django/utils/functional.py", line 235, in inner
    return func(self._wrapped, *args)
AttributeError: 'VisitingCustomer' object has no attribute '_meta'
```

It is important to me to allow user, because some users register as partner on our site, which don't buy anything from our shop.

**Comment by jrief** [comment]
This bug has been fixed in PR #583
Are you using HEAD-master, or 0.10.2?

**Comment by vivazzi** [comment]
I'm using django-shop==0.10.2. In HEAD-master I see merged PR, but there is not in 0.10.2.
I thing this PR doesn't in 0.10.2. I think we need next release, need not we?

**Comment by jrief** [comment]
can we close this?
